File: collect_app/src/main/java/org/odk/collect/android/views/DecoratedBarcodeView.java
package org.odk.collect.android.views;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

public class DecoratedBarcodeView extends com.journeyapps.barcodescanner.DecoratedBarcodeView {

    public static boolean test;

    public DecoratedBarcodeView(Context context) {
        super(context);
        initialize();
    }

    public DecoratedBarcodeView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initialize();
    }

    public DecoratedBarcodeView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initialize();
    }

    private void initialize() {
        if (test) {
            setVisibility(View.INVISIBLE);
        }
    }
}

File: app/src/main/java/org/odk/collect/android/views/DecoratedBarcodeView.kt
package org.odk.collect.android.views

import android.content.Context
import android.util.AttributeSet
import com.journeyapps.barcodescanner.DecoratedBarcodeView

class DecoratedBarcodeView : DecoratedBarcodeView {
    constructor(context: Context) : super(context) {
        initialize()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initialize()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initialize()
    }

    private fun initialize() {
        if (test) {
            visibility = INVISIBLE
        }
    }

    companion object {
        var test = false
    }
}
